module/datamanager.py

```diff
@@ -121,17 +122,15 @@ def get_percentage_hosts_state(self, user=None, problem=False):
                             True to return the % of hosts in problems.
                             False by default
         """
-        # TODO: Use livestate
-        all_hosts = self.get_hosts(user=user)
-        if not all_hosts:
+        h = self.get_hosts_synthesis(None, user)
+        if not h['nb_elts']:
             return 0
 
-        problems = [h for h in all_hosts if h.is_problem and not h.is_impact]
+        known_problems = h['nb_problems']
+        if not problem:
+            known_problems = h['nb_elts'] - h['nb_problems']
```

Comment:
`not problem` is default in the function definition. So please begin by `if not problem: … else: …`

Reply:
Ok. I will change the test